lib/wibox/widget/base.lua

```diff
@@ -420,6 +420,14 @@ function base.place_widget_at(widget, x, y, width, height)
     return base.place_widget_via_matrix(widget, matrix.create_translate(x, y), width, height)
 end
 
+-- Check if `obj` can be called (either using the metacall or as a function)
+local function is_callable(obj)
+    local t = type(obj)
+    return t == "function" or (
+        t == "table" and getmetatable(obj) and getmetatable(obj).__call
+    ), t
+end
+
```

Comment:
@ Commit message: Well, I like that boilerplate of having to add an explicit `widget =`. That provides at least some clarity between "this already is a widget, just use it" and "this is how you can construct a widget". But this declarative fun is your playground, so... whatever.